server/app/services/OrderService.ts

```diff
@@ -0,0 +1,88 @@
+import { inject, injectable } from 'inversify';
+import TYPES from '../constants/types';
+import { IOrder } from '../models/OrderModel';
+import { OrderRepository } from '../repository/OrderRepository';
+import { NotApprovedError, NotFoundError } from '../errors';
+import { MaterialService } from './MaterialService';
+import { IMaterial } from '../models/MaterialModel';
+import { materialCost, materialTypes } from '../entities/Material';
+
+@injectable()
+export class OrderService {
+  constructor(
+    @inject(TYPES.OrderRepository) private orderRepo: OrderRepository,
+    @inject(TYPES.MaterialService) private materialService: MaterialService
+  ) {}
+
+  public async getOrders(): Promise<IOrder[]> {
+    return this.orderRepo.getList();
+  }
+
+  public async createOrder(body: IOrder): Promise<IOrder> {
+    this.validateMaterial(body.materialType);
+
+    body.cost = Object.values(materialCost[body.materialType])[0];
+    return this.orderRepo.create(body);
+  }
+
+  public async updateOrder(id: string, body: IOrder): Promise<IOrder> {
+    this.validateMaterial(body.materialType);
+
+    body.cost = Object.values(materialCost[body.materialType])[0];
+    const updateOrder = await this.orderRepo.update(id, body);
+    if (!updateOrder) {
+      throw new NotFoundError(`Order with id ${id} was not found`);
+    }
+
+    return updateOrder;
+  }
+
+  public async deleteOrder(body: IOrder): Promise<IOrder | null> {
+    const deletedOrder = await this.orderRepo.delete(body.id);
+    if (!deletedOrder) {
+      throw new NotFoundError(`Order with id ${body.id} was not found`);
+    }
+
+    return deletedOrder;
+  }
+
+  public async findById(id: string): Promise<IOrder> {
+    const order = await this.orderRepo.findById(id);
+    if (!order) {
+      throw new NotFoundError(`Order with id ${id} was not found`);
+    }
+
+    return order;
+  }
+
+  public async approveOrder(body: IOrder): Promise<IOrder> {
+    if (body.status !== 'Approved') {
+      throw new NotApprovedError(`Order with id ${body.id} was not approved`);
```

Comment:
is it better if we use the orderID instead of the id of the document since we don't know it if we don't make a get call to get all orders and their document id?

What do you think

Reply:
From the frontend perspective, I think it's a must we have to make a call to get all orders (to render the table) and from there we can render the approve/deny buttons as well, what do you think

Reply:
so we need to keep track of the order id and the document id

Reply:
yes, we can do that

Reply:
Wait no, maybe we should just use the document ID given by MongoDB instead of created our own ID?

Reply:
naasir is user document id

Reply:
I am asking if we should use order id instead of document id to find orders in db

Reply:
I think document ID should be sufficient

Reply:
ok nice